src/expr/node_value.h

```diff
@@ -415,6 +424,13 @@ inline void NodeValue::inc() {
   // FIXME multithreading
   if(__builtin_expect( ( d_rc < MAX_RC ), true )) {
     ++d_rc;
```

Comment:
Since this is a heavily-used function, I think it's worth being careful so we don't introduce an extra compare in the common case.  How about this:
if (__builtin_expect( ( d_rc < MAX_RC - 1), true)) {
  ++d_rc;
} else if ( (d_rc == MAX_RC - 1) ) {
  ++d_rc;
  Assert...
  mark...
}